orchagent/portsorch.cpp

```diff
@@ -2472,7 +2486,12 @@ void PortsOrch::doPortTask(Consumer &consumer)
                         }
                     }
 
-                    if (!initPort(get<0>(it->second), get<4>(it->second), it->first))
+                    /* Skip initializing recirc port for now because the current SAI implementation of some vendors
+                     * have limiited support for recirc port. This check can be removed once SAI implementation
+                     * is enhanced/changed in the future.
+                     */
+                    if (m_recircPortRole.find(get<0>(it->second)) == m_recircPortRole.end() &&
+                        !initPort(get<0>(it->second), get<4>(it->second), it->first))
```

Comment:
why we need to do special check here ?  
Can we just go to function initPort() and inside this function do PortRole Check and if it is Inband or Recirc call the new API doProcessRecircPort().  This way code port init code is in common place

Reply:
The reason that we had a special case for recirc port is because the SAI implementation we tested has limited support for recirc port. For example, the only support port attribute for recirc port is lane set. Therefore, initPort() failed on recirc port. 

We can process recirc port inside initPort() by calling doProcessRecircPort(). I will look into this change.

Reply:
Moved recirc port processing into initPort() function.